tools/build-kani/src/sysroot.rs

```diff
@@ -56,35 +56,42 @@ pub fn kani_playback_lib() -> PathBuf {
 }
 
 /// Returns the path to where Kani's pre-compiled binaries are stored.
-pub fn kani_sysroot_bin() -> PathBuf {
+fn kani_sysroot_bin() -> PathBuf {
     path_buf!(kani_sysroot(), "bin")
 }
 
 /// Build the `lib/` folder and `lib-playback/` for the new sysroot.
 /// - The `lib/` folder contains the sysroot for verification.
 /// - The `lib-playback/` folder contains the sysroot used for playback.
-pub fn build_lib() -> Result<()> {
-    build_verification_lib()?;
-    build_playback_lib()
+pub fn build_lib(bin_folder: &Path) -> Result<()> {
+    let compiler = bin_folder.join("kani-compiler");
+    build_verification_lib(&compiler)?;
+    build_playback_lib(&compiler)
```

Comment:
Do we discuss this in our developer documentation? If not, I think it'd be nice to add an overview there in the future.

Reply:
Do you mean discuss the build process?

Reply:
Not the build process itself, but the libraries which are built and their purpose.

Reply:
Sounds good. I want to eventually document Kani's architecture too.